File: FinalProject/app/src/main/java/com/example/finalproject/items/Food.java
package com.example.finalproject.items;

public class Food extends Item {
    private int manaRecovery, healthRecovery, weight;

    public Food(int manaRecovery, int healthRecovery, int weight) {
        this.manaRecovery = manaRecovery;
        this.healthRecovery = healthRecovery;
        this.weight = weight;
    }

    public int getManaRecovery() {
        return manaRecovery;
    }

    public void setManaRecovery(int manaRecovery) {
        this.manaRecovery = manaRecovery;
    }

    public int getHealthRecovery() {
        return healthRecovery;
    }

    public void setHealthRecovery(int healthRecovery) {
        this.healthRecovery = healthRecovery;
    }

    public int getWeight() {
        return weight;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/items/Food.kt
package com.example.finalproject.service.classes.items

class Food(
    name: String,
    id: Int,
    costSell: Int,
    costBuy: Int,
    category: Int,
    rarity: Int,
    val manaRecovery: Double,
    val healthRecovery: Double,
    val weight: Double,
) : Item(
    name = name,
    id = id,
    costSell = costSell,
    costBuy = costBuy,
    rarity = rarity,
    category = category
)